src/Exception/DiscoveryFailedException.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Http\Discovery\Exception;
+
+/**
+ * Thrown when all discovery strategies fails to find a resource.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+final class DiscoveryFailedException extends \RuntimeException
+{
+    /**
+     * @var array
+     */
+    private $exceptions;
+
+    /**
+     * @param $exceptions
+     */
+    public function __construct($message, array $exceptions = [])
+    {
+        $this->exceptions = $exceptions;
+
+        parent::__construct($message, 0, array_shift($exceptions));
```

Comment:
This is a hack to make sure we at least get one exception in our exception change. Without this you will never see the "Puli is not installed"- exception.